File: src/main/java/com/codeperfection/shipit/repository/ShippingRepository.java
package com.codeperfection.shipit.repository;

import com.codeperfection.shipit.entity.Shipping;
import com.codeperfection.shipit.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface ShippingRepository extends JpaRepository<Shipping, UUID> {

    Page<Shipping> findByUser(User user, Pageable pageable);

    Optional<Shipping> findByUuidAndUser(UUID uuid, User user);
}

File: src/main/kotlin/com/codeperfection/shipit/repository/ShippingRepository.kt
package com.codeperfection.shipit.repository

import com.codeperfection.shipit.entity.Shipping
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface ShippingRepository : JpaRepository<Shipping, UUID> {

    fun findByUserId(userId: UUID, pageable: Pageable): Page<Shipping>

    fun findByIdAndUserId(id: UUID, userId: UUID): Shipping?
}
